internal/core/string.rs

```diff
@@ -459,6 +504,158 @@ pub(crate) mod ffi {
         }
     }
 
+    #[no_mangle]
+    pub unsafe extern "C" fn slint_shared_string_from_number_fixed(
+        out: *mut SharedString,
```

Comment:
```suggestion
        out: &mut SharedString,
```

The caller in cpp.rs initialize the slint::SharedString
It is then wrong to use core::ptr::write. It should simply be `*out = ....`

This is different from the slint_shared_string_from_number because that is used in the constructor on uninitialized data. 
So no unsafe needed here.

Same for slint_shared_string_from_number_precision